### PR TITLE
fix: warehouse resource

### PR DESCRIPTION
**Pull request by sivchari**
The following plan was output.

```terraform
  ~ resource "snowflake_warehouse" "query-2xlarge" {
        id                                  = "DEV_GARNET_WH_QUERY_2XLARGE"
        name                                = "DEV_GARNET_WH_QUERY_2XLARGE"
      ~ warehouse_size                      = "2XLARGE" -> "XXLARGE"
        # (13 unchanged attributes hidden)
    }
```

I fixed warehouse size definition and fixed Update/Delete resource to save the latest tfstate.

## Test Plan

* [ ] acceptance tests

### CODE REVIEW

**Review by sfc-gh-ngaberel — APPROVED** [review]

**Comment by sivchari** [comment]
Please rerun 🙏

**Comment by sfc-gh-ngaberel** [comment]
@sivchari sorry we were having issues with our CI infrastructure yesterday, it's back to normal now, I should be able to merge this today. :)

**Comment by sivchari** [comment]
@sfc-gh-swinkler @sfc-gh-ngaberel 

Can you merge this ? Thanks.

File: pkg/resources/warehouse.go [hunk 1]
```diff
@@ -435,5 +435,5 @@ func DeleteWarehouse(d *schema.ResourceData, meta interface{}) error {
 		return err
 	}
 
-	return nil
+	return ReadWarehouse(d, meta)
```
Comment (sfc-gh-swinkler) [inline pkg/resources/warehouse.go:438]: theres no reason to read the warehouse after dropping it.

**Review by sfc-gh-swinkler — CHANGES_REQUESTED** [review]

**Comment by sfc-gh-asawicki** [comment]
Hi @sivchari! Will you introduce the change that @sfc-gh-swinkler has requested?

**Comment by sfc-gh-asawicki** [comment]
Closign because warehouse resource was deeply redesigned in v0.93.0.

**Pull request closed without merging**